06-express-and-layered-architecture/src/cart/cart.controller.ts

```diff
@@ -0,0 +1,74 @@
+import express from "express";
+import * as cartService from "./cart.service";
+import * as orderService from "./order/order.service";
+import joi from "joi";
+import { calculateTotal } from "./middleware/cart.middleware";
+import { checkIfUserExists } from "./middleware/auth.middleware";
+
+const cartUpdateSchema = joi.object({
+  productId: joi.string().required(),
+  count: joi.number().required(),
+});
+
+const router = express.Router();
+router.use("/api/profile/cart", checkIfUserExists);
+
+router.get("/api/profile/cart", async (req, res) => {
+  const userId = res.locals.user;
```

Comment:
It is a bit weir that you are accessing the user on res.locals, while the auth middleware assigns it to `req.user`. I am not sure why this works, but for consitency's sake you should just get the userId from `req.user`.